test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -33,7 +33,11 @@
     `(do ~@body)
     (let [case-versions (remove keyword? (take-nth 2 bindings))]
       `(doseq ~bindings
-         (testing (str "Testing case " '~case-versions)
+         (testing (str "Testing case: " (clojure.string/join
+                                          ", "
+                                          (map #(str %1 ": " %2)
+                                               '~case-versions
+                                               (list ~@case-versions))))
```

Comment:
perhaps the chainable prose with a ```(pr-str )``` hardened variant of the original implementation:
```clojure
(defmacro dotestseq [bindings & body]
  (if-not (seq bindings)
    `(do ~@body)
    (let [case-versions (remove keyword? (take-nth 2 bindings))]
      `(doseq ~bindings
         (testing (str "with " (clojure.string/join
                                 ", "
                                 (map #(str (pr-str %1) ": " (pr-str %2))
                                      '~case-versions
                                      (list ~@case-versions))))
           ~@body)))))
```
which produces this output:
```
FAIL in (some-test) (testutils.clj:47)
check something with {:keys [a b]}: {:keys [10 20]}, [c d]: [3 4]
expected: (= 13 [a b c d])
  actual: (not (= 13 [10 20 3 4]))
```

@rbrw what are your thoughts?